### PR TITLE
Users should be able to choose which servers are for reading, and which are for writing.

### PR DESCRIPTION
**Pull request by nostrdev-com**
Adjusted kind 10096 in [NIP-96](https://github.com/nostr-protocol/nips/blob/master/96.md#selecting-a-server) to be more closely aligned with [NIP-65](https://github.com/nostr-protocol/nips/blob/master/65.md)

Use case - in our application ([sigit.io](https://sigit.io)) we allow users to store their data on their own blossom server(s).

The problem is that they may have some servers which are readonly (eg if their subscription expired), or perhaps they have upload limits and would like to stop writing to a particular server (but still read from it).  Or would just like more control over their storage defaults in general.

### CODE REVIEW

**Comment by staab** [comment]
I don't see this as equivalent to NIP 65, since the purpose of read relays there is to advertise where people should put notes that mention the user. I'm not sure how a media upload would be "intended" for a user. If it is, it should probably be accompanied by some nostr event that gets put in the correct place according to NIP 65 and contains a media server hint.

**Comment by nostrdev-com** [comment]
Our use case is a mechanism that allows client apps to know to which file servers they should be saving (or fetching) arbitrary files.

Whilst many Nostr apps may suggest / hard code a default file server, individual users may prefer to use (or to upload in addition) their own file servers.

Is there another option / NIP configuration by which a client app can determine which file servers can be used for reading / writing files?

**Comment by staab** [comment]
I haven't been super involved with blossom or nip 96, but what I was hoping for is what we have with regular events, i.e. pubkey/relay hints. A hash on its own isn't enough to find a file, and the odds of it being on a random media server (even a very popular one) are relatively low. NIP 94 includes `url`, `fallback`, and `service` tags, as well as the pubkey of the person creating the event, which you can use to find that person's 10096 "write" servers. NIP 92 `imeta` has all the same fields, and should probably include a `pubkey` indicating the original author.

Long story short, this use case would be best saved by following a `pubkey` to a `10096` list, or failing that by following urls/services (which sort of defeats the purpose of hash-based file storage).

With all that said I realized you're talking specifically about NIP 96 servers, which I could see acting like proxies, in which case something like this could make sense, but only if cache misses get resolved on the server side.
